pinot-core/src/main/java/com/linkedin/pinot/core/query/aggregation/groupby/AggregationGroupByOperatorService.java

```diff
@@ -243,9 +243,19 @@ public AggregationGroupByOperatorService(List<AggregationInfo> aggregationInfos,
     return aggregationResults;
   }
 
-  private Serializable formatValue(Serializable value) {
-    return (value instanceof Float || value instanceof Double) ? String.format(Locale.US, "%1.5f", value)
-        : value.toString();
+  /**
+   * Format the {@link Serializable} value into {@link String}.
+   * <p>For {@link Double} value, format it into the form #####.#####.
+   *
+   * @param value value to be formatted.
+   * @return formatted value.
+   */
+  public static String formatValue(Serializable value) {
+    if (value instanceof Double) {
+      return String.format(Locale.US, "%1.5f", (Double) value);
```

Comment:
This format only works for floating point numbers.
Changed the format to "###################0.0#########" which is consistent with our selection response. This way we can differentiate integral numbers and floating point numbers.
My only concern is that this will change the response format. (E.g. 1.00000 --> 1.0)